### PR TITLE
en/models: Document models for tesseract, ocropy, calamari

### PR DESCRIPTION
**Pull request by kba**

### CODE REVIEW

File: site/en/models.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Models for OCR-D processors
+
+OCR engines rely on pre-trained models for their recognition. Every engine has
+its own internal format(s) for models, some support central storage of models
```
Comment (EEngl52) [inline site/en/models.md:4]:
```suggestion
its own internal format(s) for models. Some support central storage of models
```

File: site/en/models.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+In the meantime, this guide will show you, for each OCR engine:
+
+  * which types of models are supported
+  * where to store models locally
+  * which currently available models we recommend
+  * how to invoke the resp. OCR-D wrapper for the engine with a specific model
+
+## Tesseract / ocrd_tesserocr
+
+Tesseract models are single files witha `.traineddata` extension.
```
Comment (EEngl52) [inline site/en/models.md:20]:
```suggestion
Tesseract models are single files with a `.traineddata` extension.
```

File: site/en/models.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+# Use the deu and frk models
+ocrd-tesserocr-recognize -I OCR-D-SEG-LINE -O OCR-D-OCR-TESS -p '{"model": "deu+frk"}'
+# Use the script/Fraktur model
+ocrd-tesserocr-recognize -I OCR-D-SEG-LINE -O OCR-D-OCR-TESS -p '{"model": "script/Fraktur"}'
+```
+
+## Ocropy / ocrd_cis
+
+An Ocropy model is simply the neural network serialized as with Python's pickle
+mechanism and are generally distributed in a gzipped form, with an `.pyrnn.gz`
```
Comment (EEngl52) [inline site/en/models.md:60]:
```suggestion
mechanism and is generally distributed in a gzipped form, with an `.pyrnn.gz`
```

File: site/en/models.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+extension.
+
+Ocropy has a rather convoluted algorithm to look up models, so we recommend you
+explicitly set the `OCROPUS_DATA` variable to point to the directory with
+ocropy's models. E.g. if you intend to store your models in `$HOME/ocropus-models`, add the following
+to you `$HOME/.bashrc`: `export OCROPUS_DATA=$HOME/ocropus-models`.
```
Comment (EEngl52) [inline site/en/models.md:66]:
```suggestion
to your `$HOME/.bashrc`: `export OCROPUS_DATA=$HOME/ocropus-models`.
```

File: site/en/models.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  * [frk](https://github.com/tesseract-ocr/tessdata_fast/raw/master/frk.traineddata)
+  * [script/Latin](https://github.com/tesseract-ocr/tessdata_fast/raw/master/script/Fraktur.traineddata)
+  * [script/Fraktur](https://github.com/tesseract-ocr/tessdata_fast/raw/master/script/Fraktur.traineddata)
+  * [@stweil's GT4HistOCR model](https://ub-backup.bib.uni-mannheim.de/~stweil/ocrd-train/data/fast/Fraktur_50000000.334_450937.traineddata)
+
+OCR-D's Tesseract wrapper,
+[ocrd_tesserocr](https://github.com/OCR-D/ocrd_tesserocr) and more
+specifically, the `ocrd-tesserocr-recognize` processor, expects the name of the
+model(s) to be provided as the `model` parameter. Multiple models can be
+combined by concatenating with `+`:
```
Comment (bertsky) [inline site/en/models.md:48]:
```suggestion
combined by concatenating with `+` (which generally improves accuracy but always slows processing):
```

File: site/en/models.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  * [@jze's fraktur.pyrnn.gz](https://github.com/jze/ocropus-model_fraktur/raw/master/fraktur.pyrnn.gz) (save as `fraktur-jze.pyrnn.gz`)
+  * [@chreul's  LatinHist.pyrnn.gz](https://github.com/chreul/OCR_Testdata_EarlyPrintedBooks/raw/master/LatinHist-98000.pyrnn.gz)
+
+
+To use a specific model with OCR-D's ocropus wrapper in [ocrd_cis](https://github.com/cisocrgroup/ocrd_cis) and more specifically, the `ocrd-cis-ocropy-recognize` processor, use the `model` parameter:
+
+```sh
+ocrd-cis-ocropy-recognize -I OCR-D-SEG-LINE -O OCR-D-OCR-OCRO -p '{"model": "fraktur-jze.pyrnn.gz"}'
+```
+
```
Comment (EEngl52) [inline site/en/models.md:83]: ocropy does not allow to combine several models (as opposed to Tesseract)?

File: site/en/models.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+`ocrd_all`:
+
+  * [@mike-gerber's GT4HistOCR model](https://qurator-data.de/calamari-models/GT4HistOCR/model.tar.xz)
+
+To use a specific model with OCR-D's calamari wrapper
+[ocrd_calamari](https://github.com/OCR-D/ocrd_calamari) and more specifically,
+the `ocrd-calamari-recognize` processor, use the `checkpoint` parameter:
+
+```sh
+ocrd-calamari-recognize -I OCR-D-SEG-LINE -O OCR-D-OCR-CALA -p '{"checkpoint": "/path/to/mode/*.ckpt.json"}'
```
Comment (EEngl52) [inline site/en/models.md:104]:
```suggestion
ocrd-calamari-recognize -I OCR-D-SEG-LINE -O OCR-D-OCR-CALA -p '{"checkpoint": "/path/to/model/*.ckpt.json"}'
```

File: site/en/models.md (hunk 6 above)
Comment (kba) [reply site/en/models.md:83]: No.

File: site/en/models.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+saving to the right location or by running `make install-models-tesseract` when
+using `ocrd_all`:
+
+  * [equ](https://github.com/tesseract-ocr/tessdata_fast/raw/master/equ.traineddata)
+  * [osd](https://github.com/tesseract-ocr/tessdata_fast/raw/master/osd.traineddata)
+  * [eng](https://github.com/tesseract-ocr/tessdata_fast/raw/master/eng.traineddata)
+  * [deu](https://github.com/tesseract-ocr/tessdata_fast/raw/master/deu.traineddata)
+  * [frk](https://github.com/tesseract-ocr/tessdata_fast/raw/master/frk.traineddata)
+  * [script/Latin](https://github.com/tesseract-ocr/tessdata_fast/raw/master/script/Fraktur.traineddata)
+  * [script/Fraktur](https://github.com/tesseract-ocr/tessdata_fast/raw/master/script/Fraktur.traineddata)
```
Comment (bertsky) [inline site/en/models.md:41]: I don't know whether this detail should be covered in the guide, but when installing models via dpkg, there's no `script/` prefix...

Comment (kba) [reply site/en/models.md:41]: Good point, I'll add a note on that.

File: site/en/models.md (hunk 6 above)
Comment (bertsky) [reply site/en/models.md:83]: This desideratum is called _voting_ and has not been implemented in either pyrnn or clstm backends (and thus, both Ocropus and Kraken) to my knowledge. Only Tesseract and Calamari engines have it.

File: site/en/models.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+## Calamari / ocrd_calamari
+
+Calamari models are Tensorflow model directories. For distribution, this
+directory is usually packed to a tarball or ZIP file. Once downloaded, these
+containers must be unpacked to a directory again.
+
+As calamari does not have a model discovery setup, you must always provide the
+path with a wildcard listing all `*.ckpt.json` ("checkpoint") files.
```
Comment (bertsky) [inline site/en/models.md:91]: ...`absolute path with a literal (unexpanded) wildcard`...

Comment (kba) [reply site/en/models.md:91]: It's not necessarily an absolute path. The wildcard could only be picked up by the shell if the JSON expression is completely escaped except for the asterisk.

I.e.

```
ocrd-calamari-recognize -p '{"checkpoint": "model/*.cpkt.json"}'
ocrd-calamari-recognize -p "{\"checkpoint\": \"model/*.cpkt.json\"}"
```

would not trigger the issue, only

```
ocrd-calamari-recognize -p \{\"checkpoint\": \"model/*.cpkt.json\"\}
```

might, but if the user is able to escape all other characters, I'd assume they would also escape the asterisk.

Comment (bertsky) [reply site/en/models.md:91]: AFAICT currently relative path names just won't work, because in the scope of the processor CWD is the workspace directory, which is really hard to relate to package installations.

What I meant by unexpanded wildcard is that the expansion **must** happen in calamari, you cannot pass the _result_ of a regular shell expansion (which would be a space-separated list).
